include/nighthawk/common/request_source_plugin.h

```diff
@@ -0,0 +1,50 @@
+#pragma once
+
+#include "envoy/api/api.h"
+#include "envoy/common/pure.h"
+#include "envoy/config/typed_config.h"
+
+#include "nighthawk/common/request_source.h"
+
+namespace Nighthawk {
+
+/**
+ * An interface for different RequestSource plugins.
+ * Uses a plugin-specific config proto.
+ */
+class RequestSourcePlugin : RequestSource {
+public:
+  RequestSourcePlugin(Envoy::Api::Api& api) : api_(api) {}
+
+protected:
+  Envoy::Api::Api& api_;
```

Comment:
This seems like implementation in an interface. Consider creating a common base class which derives from the interface for this?

Reply:
I just got rid of this.